src/Pate/Equivalence.hs

```diff
@@ -468,6 +564,9 @@ getRegionEquality sym memEq memO memP = case memEq of
   MemEqAtRegion stackRegion -> MT.memEqAtRegion sym stackRegion memO memP
   MemEqOutsideRegion stackRegion -> MT.memEqOutsideRegion sym stackRegion memO memP
 
+newtype RegisterCondition sym arch =
+  RegisterCondition { regCondPreds :: MM.RegState (MM.ArchReg arch) (Const (W4.Pred sym)) }
+
 -- | Compute a predicate that is true iff the two given states are equal with respect to
```

Comment:
It seems to be more structured than a `Pred` now. Is it the case that the `Pred` for each register in the `RegisterCondition` satisfies this statement now?